graphql-dgs-spring-boot-oss-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/autoconfig/DgsConfigurationProperties.kt

```diff
@@ -29,5 +29,7 @@ import org.springframework.boot.context.properties.bind.DefaultValue
 @Suppress("ConfigurationProperties")
 data class DgsConfigurationProperties(
     /** Location of the GraphQL schema files. */
-    @DefaultValue(DEFAULT_SCHEMA_LOCATION) val schemaLocations: List<String>
+    @DefaultValue(DEFAULT_SCHEMA_LOCATION) val schemaLocations: List<String>,
+    /** Enable Introspection */
+    @DefaultValue("true") val introspection: Boolean
```

Comment:
I personally prefer a more explicit name, such as `introspection.enabled` as well as docs that express the caveats. e.g.
```
/**
 * Setting this value to `false` will prevent Introspection queries from being performed.
 * Note that this is against GraphQL Specification, that said,  some production systems want this lock down in place.
 */
```
---

If we treat properties as APIs, which imo we should, this looks like a _good enough_ first approach. At the same time it got me thinking. Are there any other options that we would like to configure for _field visibility_? What if someone wants to use `BlockedFields` instead of `NoIntrospectionGraphqlFieldVisibility`?

Currently the `GraphQLCodeRegistry` accepts only one kind of _field visibility_, developers can implement their own and out of the box we have:

- DefaultGraphqlFieldVisibility
- NoIntrospectionGraphqlFieldVisibility
- BlockedFields

Then _visibility_ could be treated as a _bean_ and be a `@ConditionalOnMissingBean`. By _default_ the `DefaultGraphqlFieldVisibility` bean will be provided by it can be turned off by enabling a different _kind_ such as the `NoIntrospectionGraphqlFieldVisibility`, or provide your own implementation of a `GraphqlFieldVisibility` and express it as a bean. The latter will help if developers are interested in using `BlockedFields` or express their own logic.

@paulbakker @srinivasankavitha wdyt?